clcache.py

```diff
@@ -56,6 +56,7 @@
 #   value: key in the cache, under which the object file is stored
 Manifest = namedtuple('Manifest', ['includeFiles', 'includesContentToObjectMap'])
 
+CompilerArtifacts = namedtuple('CompilerArtifacts', ['objectFilePath', 'compilerStdout', 'compilerStderr'])
```

Comment:
> That reminds me though that using dir is a builtin function (i.e. it's imported by default since 'dir' in dir(__builtins__) is true), maybe not a good choice for the Cache.dir field.

Looks like it is not an issue when used as a member instead of a free symbol. There is not naming clash between Cache.dir and dir.

However, Cache.directory might be equally good.